### DISCUSSION TITLE
Ideas for Reducing Render-Blocking CSS in Next.js

### DISCUSSION THREAD

**Discussion by snehakhobragade90** [vercel/next.js#70526] [Help]
### Summary

I'm looking for some ideas on how to reduce Total Blocking Time on our Next.js pages caused by render-blocking CSS. 
The main concern is that some CSS is blocking the main thread, which impacts performance.

One approach I'm considering is lazy-loading CSS for below-the-fold components, which would defer their styles until needed, helping to unblock the main thread earlier. However, since Next.js already handles CSS chunking by default, I’m not entirely sure how much control I have over this via configuration to implement lazy-loading for specific styles. Does anyone have experience with fine-tuning this behavior or suggestions on how to further optimize CSS delivery for below-the-fold content?

### Additional information

Webpagetest for all of our pages shows we have around 25 externally-referenced CSS files which are blocking page rendering.

These are the webpack configs we are using at the moment to ensure that our LESS styles from other packages are transformed into scoped, optimized CSS to be supported by Next.

```js
config.module.rules.push({
      test: /\.module\.less$/,
      use: [
        {
          loader: require.resolve('style-loader'),
        },
        {
          loader: require.resolve('css-loader'),
          options: {
            sourceMap: true,
            modules: true,
          },
        },
        {
          loader: require.resolve('postcss-loader'),
          options: {
            postcssOptions: {
              sourceMap: true,
            },
          },
        },
        {
          loader: require.resolve('less-loader'),
          options: {
            lessOptions: {
              plugins: [new LessPluginNpmImport()],
            },
          },
        },
      ],
    });
```
```


### Example

_No response_

**Comment by romhenri** [comment]
Lazy Load Below the Fold CSS should be a nice solution.

Another idea would be Critical CSS Inlining, it involves extracting and inlining only the CSS styles needed to render the page's visible content (above the fold) directly into the <head> of the HTML. This allows the browser to start rendering the page as quickly as possible without waiting for all the CSS files to fully load. 

See:
https://purgecss.com/
or
https://github.com/addyosmani/critical
